VenafiPS/Public/Test-TppToken.ps1

```diff
@@ -0,0 +1,116 @@
+<#
+.SYNOPSIS
+Test if a Tpp token is valid
+
+.DESCRIPTION
+Use the TPP API call 'Authorize/Verify' to test if the current token is valid.
+
+.PARAMETER AuthServer
+Auth server or url, venafi.company.com or https://venafi.company.com.
+If just the server name is provided, https:// will be appended.
+
+.PARAMETER AccessToken
+Access token retrieved outside this module.  Provide a credential object with the access token as the password.
+
+.INPUTS
+Accesstoken
+
+.OUTPUTS
+Boolean or PSCustomObject with the following properties:
+    AuthUrl
+    AccessToken
+    RefreshToken
+    Scope
+    Identity
+    TokenType
+    ClientId
+    Expires
+
+.EXAMPLE
+Test-TppToken -AuthServer 'mytppserver.example.com' -AccessToken $cred
+Verify that current accesstoken is valid
+
+.LINK
+http://VenafiPS.readthedocs.io/en/latest/functions/Test-TppToken/
+
+.LINK
+https://github.com/gdbarron/VenafiPS/blob/main/VenafiPS/Code/Public/Test-TppToken.ps1
+
+.LINK
+https://docs.venafi.com/Docs/20.4SDK/TopNav/Content/SDK/AuthSDK/r-SDKa-GET-Authorize-Verify.php?tocpath=Auth%20SDK%20reference%20for%20token%20management%7C_____13
+
+#>
+function Test-TppToken {
+
+    [CmdletBinding()]
+    param (
+        [Parameter(Mandatory)]
+        [ValidateScript( {
+                if ( $_ -match '^(https?:\/\/)?(((?!-))(xn--|_{1,1})?[a-z0-9-]{0,61}[a-z0-9]{1,1}\.)*(xn--)?([a-z0-9][a-z0-9\-]{0,60}|[a-z0-9-]{1,30}\.[a-z]{2,})$' ) {
+                    $true
+                }
+                else {
+                    throw 'Please enter a valid server, https://venafi.company.com or venafi.company.com'
+                }
+            }
+        )]
+        [Alias('Server')]
+        [string] $AuthServer,
+
+        [Parameter(Mandatory)]
+        [PSCredential] $AccessToken,
+
+        [Parameter()]
+        [VenafiSession] $VenafiSession = $script:VenafiSession
+    )
+
+    begin {
+        $authType = $VenafiSession.Validate()
```

Comment:
`$authType = $VenafiSession.Validate('token')`, adding 'token' will cause it to be validated and fail otherwise.  you only want to run this if $venafisession was provided/has a value

Reply:
Fixed in rework